notebook/nbextensions.py

```diff
@@ -299,118 +355,300 @@ def _config_file_name_default(self):
+        }}, "Install to the user's Jupyter directory"
+    ),
+    "sys-prefix" : ({
+        "ToggleNBExtensionApp" : {
+            "sys_prefix" : True,
+        }}, "Use sys.prefix as the prefix for installing nbextensions"
+    ),
+    "py" : ({
+        "ToggleNBExtensionApp" : {
+            "python" : True,
+        }}, "Install from a Python package"
+    ),
+}
+
+class ToggleNBExtensionApp(BaseNBExtensionApp):
+    
+    name = "jupyter nbextension enable/disable"
     version = __version__
-    description = "Configure an nbextension to be automatically loaded"
+    description = "Enable/disable an nbextension using frontend configuration files."
 
     section = Unicode('notebook', config=True,
-          help=("Which config section to add the extension to. "
-                "'common' will affect all pages.")
+          help="""Which config section to add the extension to, 'common' will affect all pages."""
     )
 
-    aliases = {'section': 'EnableNBExtensionApp.section',
-              }
+    aliases = {'section': 'ToggleNBExtensionApp.section'}
+    flags = _toggle_flags
+
+    python = Bool(False, config=True, help="Install from a Python package")
+    user = Bool(False, config=True, help="Whether to do a user install")
+    sys_prefix = Bool(False, config=True, help="Use the sys.prefix as the prefix")
+
+    _toggle_value = None
 
     def _config_file_name_default(self):
         return 'jupyter_notebook_config'
 
-    def enable_nbextension(self, name):
-        # Local import to avoid circular import issue on Py 2
-        from .services.config import ConfigManager
-        cm = ConfigManager(parent=self, config=self.config)
-        cm.update(self.section, {"load_extensions": {name: True}})
-
+    def _toggle_nbextension(self, section, require):
+        config_dir = os.path.join(_get_config_dir(user=self.user, sys_prefix=self.sys_prefix), 'nbconfig')
+        cm = BaseJSONConfigManager(parent=self, config_dir=config_dir)
+        if self._toggle_value is None and require not in cm.get(section).get('load_extensions', {}):
+            self.log.warn('{} is not enabled in section {}'.format(require, section))
+            sys.exit(1)
+        # We're using a dict as a set - updating with None removes the key
+        cm.update(section, {"load_extensions": {require: self._toggle_value}})
+    
+    def toggle_nbextension_python(self, package):
+        m, nbexts = _get_nbextension_metadata(package)
+        for nbext in nbexts:
+            section = nbext['section']
+            require = nbext['require']
+            self._toggle_nbextension(section, require)
+
+    def toggle_nbextension(self, require):
+        self._toggle_nbextension(self.section, require)
+        
     def start(self):
+
         if not self.extra_args:
-            sys.exit('No extensions specified')
+            self.log.warn('Please specify an nbextension/package to enable or disable')
+            sys.exit(1)
         elif len(self.extra_args) > 1:
```

Comment:
This check is correct, it's not the problem. It's failing because `--py` and `--sys-prefix` are not registered as aliases, so they aren't parsed at all.